lib/store/index.js

```diff
@@ -10,7 +10,8 @@ import type Kernel from "./../kernel";
 
 class Store {
   subscriptions = new CompositeDisposable();
-  @observable runningKernels = new Map();
+  @observable startingKernels: Map<string, boolean> = new Map();
+  @observable runningKernels: Map<string, Kernel> = new Map();
   @observable editor = atom.workspace.getActiveTextEditor();
```

Comment:
I like the idea of having a promise or to just give `store` a new kernel even if it's not fully connected yet. But this likely requires some refactorings that will be a lot easier once #858 is closed.